Userland/head.cpp

```diff
@@ -1,25 +1,87 @@
+#include <errno.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <LibCore/CArgsParser.h>
+
+int head(String filename, bool print_filename, int line_count);
 
 int main(int argc, char** argv)
 {
-    // FIXME: Allow setting via command-line argument.
+    CArgsParser args_parser("head");
+
+    args_parser.add_arg("n", "lines", "Number of lines to print (default 10)");
+    args_parser.add_arg("q", "Never print filenames");
+    args_parser.add_arg("v", "Always print filenames");
+
+    CArgsParserResult args = args_parser.parse(argc, (const char**)argv);
+
     int line_count = 10;
+    if (args.is_present("n")) {
+        line_count = strtol(args.get("n").characters(), NULL, 10);
+        if (errno == EINVAL) {
+            args_parser.print_usage();
+            return -1;
+        }
+    }
 
-    if (argc > 1) {
-        for (int i = 1; i < argc; ++i) {
-            if (argv[i][0] == '-') {
-                line_count = atoi(&argv[i][1]);
+    Vector<String> files = args.get_single_values();
+
+    bool print_filenames = files.size() > 1;
+
+    if (args.is_present("v")) {
+        print_filenames = true;
+    } else if (args.is_present("q")) {
+        print_filenames = false;
+    }
+
+    int rc = 0;
+
+    if (files.size() == 0) {
+        if (head("", print_filenames, line_count) != 0) {
+            rc = 1;
+        }
+    } else {
+        for (auto it = files.begin(); it != files.end(); ++it) {
+            if (head(*it, print_filenames, line_count) != 0) {
+                rc = 1;
             }
         }
     }
 
+    return rc;
+}
+
+int head(String filename, bool print_filename, int line_count)
+{
+    FILE* fd = nullptr;
```

Comment:
By convention, `fd` is a file descriptor. `fp` is typically used for `FILE*` pointers :)